src/test/java/com/lgtm/easymoney/controllers/RequestControllerTest.java

```diff
@@ -0,0 +1,366 @@
+package com.lgtm.easymoney.controllers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.lgtm.easymoney.enums.Category;
+import com.lgtm.easymoney.enums.TransactionStatus;
+import com.lgtm.easymoney.exceptions.InvalidUpdateException;
+import com.lgtm.easymoney.exceptions.ResourceNotFoundException;
+import com.lgtm.easymoney.models.Transaction;
+import com.lgtm.easymoney.payload.RequestAcceptDeclineReq;
+import com.lgtm.easymoney.payload.RequestReq;
+import com.lgtm.easymoney.payload.RequestRsp;
+import com.lgtm.easymoney.payload.ResourceCreatedRsp;
+import com.lgtm.easymoney.payload.TransactionRsp;
+import com.lgtm.easymoney.services.RequestService;
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * Unit test for TransferController.
+ * {@link com.lgtm.easymoney.controllers.RequestController}
+ * */
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(RequestController.class)
+@ContextConfiguration
+public class RequestControllerTest {
+  @Autowired
+  private MockMvc mvc;
+  @MockBean
+  private RequestService requestService;
+
+  private RequestReq requestReq;
+
+  private RequestRsp requestRsp;
+  private ResourceCreatedRsp resourceCreatedRsp;
+  private TransactionRsp transactionRsp;
+
+  private Long fromUid = 1L;
+  private Long toUid = 2L;
+
+  private Long requestId = 1L;
+  private BigDecimal amount = BigDecimal.valueOf(30.0);
+  private String description = "this is a test request";
+  private Date lastUpdateTime = new Date(20221020L);
+
+  /**
+   * Set up reusable test fixtures.
+   * */
+  @Before
+  public void setUp() {
+    // request for making request
+    requestReq = new RequestReq();
+    requestReq.setFromUid(fromUid);
+    requestReq.setToUid(toUid);
+    requestReq.setAmount(amount);
+    requestReq.setCategory(Category.PARTY);
```

Comment:
What if the client sends a string not in the enum value? The client is probably not aware of any enums defined on the server end (it just sends string), so arbitrary category input by from client cannot be simulated in this way. Perhaps better to change the corresponding payload field type to string and annotate it with `@ValueOfEnum(enumClass = Category.class)`